xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererIMX.cpp

```diff
@@ -178,25 +184,25 @@ bool CRendererIMX::RenderUpdateVideoHook(bool clear, DWORD flags, DWORD alpha)
     //CLog::Log(LOGDEBUG, "BLIT RECTS: source x1 %f x2 %f y1 %f y2 %f dest x1 %f x2 %f y1 %f y2 %f", srcRect.x1, srcRect.x2, srcRect.y1, srcRect.y2, dstRect.x1, dstRect.x2, dstRect.y1, dstRect.y2);
     g_IMXContext.SetBlitRects(srcRect, dstRect);
 
-    uint8_t fieldFmt = 0;
-    if (g_graphicsContext.IsFullScreenVideo())
+    uint8_t fieldFmt = flags & RENDER_FLAG_FIELDMASK;
+    if (flags & RENDER_FLAG_FIELDS && g_graphicsContext.IsFullScreenVideo())
     {
-      fieldFmt |= flags & RENDER_FLAG_FIELDMASK;
-      if (flags & RENDER_FLAG_FIELDS)
+      fieldFmt |= IPU_DEINTERLACE_RATE_EN;
+      if (flags & RENDER_FLAG_FIELD1)
       {
-        fieldFmt |= IPU_DEINTERLACE_RATE_EN;
-        if (flags & RENDER_FLAG_FIELD1)
-        {
-          fieldFmt |= IPU_DEINTERLACE_RATE_FRAME1;
-          // CXBMCRenderManager::PresentFields() is swapping field flag for frame1
-          // this makes IPU render same picture as before, just shifted one line.
-          // let's correct this
-          fieldFmt ^= RENDER_FLAG_FIELDMASK;
-        }
+        fieldFmt |= IPU_DEINTERLACE_RATE_FRAME1;
+        // CXBMCRenderManager::PresentFields() is swapping field flag for frame1
+        // this makes IPU render same picture as before, just shifted one line.
+        // let's correct this
+        fieldFmt ^= RENDER_FLAG_FIELDMASK;
```

Comment:
I find it (personally) odd - and the IPU deint code for BOB too (as it is internally doing the same again - so finally rendering same frame, starting at line 2 instead of 1).

On the other hand the GLES code is handling (this) same output from RenderManager and produced picture is fine. At least on v14 (when IMX had GLES rendering too - I reused AML code for IMX back than - and that code is still used for AML).

... someone would have already realised if similar issue would exists for other implementations (I think).